Backend/src/main/java/com/abhinternship/auctionapp/repository/ProductRepository.java

```diff
@@ -39,4 +36,6 @@
     List<Double> getAllByOrderByHighestBid();
 
     List<Product> getAllByOrderByEndDateAsc(Pageable pageable);
-}
\ No newline at end of file
+
+    List<Product> getAllByUser(User user, Pageable pageable);
```

Comment:
Rename all these get methods appropriately, and pass just the data needed to get the object (user id).